src/main/java/io/fabric8/jenkins/openshiftsync/BuildStateUpdater.java

```diff
@@ -43,84 +39,56 @@
 import static java.util.logging.Level.SEVERE;
 
 /**
- * Listens to Jenkins Job build Step then ensure there's a suitable {@link Build} object in
- * OpenShift that's updated correctly with the Job {@link Run} status
+ * updates the build resources status in openshift for a given
+ * jenkins run resource.
  */
-@Extension
-public class BuildSyncRunListener implements GraphListener {
 
-  public static final String JENKINS_ROOT_URL_ENV_VAR = "JENKINS_ROOT_URL";
-  private static final Logger logger = Logger.getLogger(BuildSyncRunListener.class.getName());
+public class BuildStateUpdater {
 
-  private String namespace;
-  private ExecutorService updateExecutor;
-  private OpenShiftClient openShiftClient;
+  private static BuildStateUpdater instance;
+  private final transient ExecutorService threadExecutor;
 
-  public BuildSyncRunListener() {
-    openShiftClient = getOpenShiftClient();
-    namespace = OpenShiftUtils.getNamespaceOrUseDefault(namespace, openShiftClient);
-    updateExecutor = Executors.newSingleThreadExecutor();
+  private BuildStateUpdater() {
+    this.threadExecutor = Executors.newSingleThreadExecutor();
```

Comment:
ah ha ... so you've factored out the namespace from this and made it a singleton

All these changes and renames will make it harder to pick this fix into upstream ... but these 2 versions of sync plugin are sooo divergent now ... merging them will require some form of detailed top down analysis

Since we are not getting reports of the original error, I think we should wait for pulling in this individual commit.

I'll mull it over here on my end, cross reference with my team here, but that is my current leaning.

Reply:
> but these 2 versions of sync plugin are sooo divergent now ... merging them will require some form of detailed top down analysis

Agreed, :(